src/w3c/templates/headers.js

```diff
@@ -7,7 +7,13 @@ export default conf => {
   const html = hyperHTML;
   return html`<div class='head'>
   ${conf.logos.map(showLogo)}
-  <h1 class='title p-name' id='title'>${conf.title}</h1>
+  
+  ${document.querySelector("h1#title, body>h1") ? (
```

Comment:
Above the `return html` line, you could have a check for `const specTitleElem = document.querySelector("h1#title, body>h1")` or just create a HTML element if it's missing... then:

```JS
const specTitleElem = document.querySelector("h1#title, body>h1") || document.createElement("h1") ;
if (specTitleElem.parentElement) { // remove it from the DOM tree, because we are going to relocate it
  specTitleElem.remove();
}
docorateSpecTitle(specTitleElem);
return html`<div class='head'>
    ${conf.logos.map(showLogo)}		    ${conf.logos.map(showLogo)}
    ${specTitleElem}
```

And `function docorateSpecTitle()` would check or add the `title` css class, `id` attribute, and whatever else is needed. 

The goal is never to repeat ourselves ([DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)) and our goal is always to make code as legible and efficient as possible... note your original code has "document.querySelector" 3x times for the same thing, which is not ideal - both because it's hard to read, and because it's very inefficient to ask the browser to perform the query.   

Makes sense? Again, I don't know if the above code I proposed works - it's probably buggy (thus we need test for it), but should be hopefully enough to go from.

Reply:
This is an awesome review! I'll make the changes accordingly. And yeah, [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).